json-schema/json-schema-macros/src/main/scala/endpoints/macros/Macros.scala

```diff
@@ -0,0 +1,208 @@
+package endpoints.macros
+
+import scala.reflect.macros.blackbox
+
+class Macros(val c: blackbox.Context) extends Utils {
+
+  import c.universe._
+
+  final def genericJsonSchemaImpl[A: c.WeakTypeTag]: c.Tree = {
+    val A = weakTypeOf[A]
+    findJsonSchema(A).getOrElse {
+      c.abort(c.enclosingPosition, s"Can't derive JsonSchema[$A]")
+    }
+  }
+
+  private def findJsonSchema(tpe: Type, allowImplicitSearch: Boolean = false): Option[c.Tree] = {
+    findPrimitive(tpe)
```

Comment:
Well, they do - thanks for hint!

However in macros tests, where we have trivial string-based interpreter, compiler initially couldn't find primitive instances through implicit search due to ambiguities caused by having `type JsonSchema[A] = String`. It can be fixed by tagging the type in test interpreter (i.e. `type JsonSchema[A] = String with Tag[A]`). OTOH direct primitive approach worked well also for trivial interpreter. No strong opinion here which one is better :)